src/Backend/Modules/Blog/Engine/Model.php

```diff
@@ -232,63 +234,55 @@ public static function deleteCategoryAllowed(int $id): bool
         );
     }
 
-    /**
-     * Deletes one or more comments
-     *
-     * @param array $ids The id(s) of the items(s) to delete.
-     */
     public static function deleteComments(array $ids): void
     {
-        // make sure $ids is an array
-        $ids = (array) $ids;
-
-        // loop and cast to integers
-        foreach ($ids as &$id) {
-            $id = (int) $id;
-        }
+        $repository = BackendModel::get('doctrine.orm.default_entity_manager')
+            ->getRepository(Comment::class);
 
-        // create an array with an equal amount of questionmarks as ids provided
-        $idPlaceHolders = array_fill(0, count($ids), '?');
+        $comments = $repository->findById($ids);
 
-        // get database
-        $database = BackendModel::getContainer()->get('database');
+        if (empty($comments)) {
+            return;
+        }
 
-        // get ids
-        $itemIds = (array) $database->getColumn(
-            'SELECT i.post_id
-             FROM blog_comments AS i
-             WHERE i.id IN (' . implode(', ', $idPlaceHolders) . ')',
-            $ids
-        );
+        $postIdsToRecalculate = [];
+        foreach ($comments as $comment) {
+            $postIdsToRecalculate[] = $comment->getPostId();
+        }
 
-        // update record
-        $database->delete('blog_comments', 'id IN (' . implode(', ', $idPlaceHolders) . ')', $ids);
+        $repository->deleteMultipleById($ids);
 
         // recalculate the comment count
-        if (!empty($itemIds)) {
-            self::reCalculateCommentCount($itemIds);
+        if (!empty($postIdsToRecalculate)) {
+            $postIdsToRecalculate = array_unique($postIdsToRecalculate);
+            self::reCalculateCommentCount($postIdsToRecalculate);
         }
     }
 
     public static function deleteSpamComments(): void
     {
-        $database = BackendModel::getContainer()->get('database');
-
-        // get ids
-        $itemIds = (array) $database->getColumn(
-            'SELECT i.post_id
-             FROM blog_comments AS i
-             WHERE status = ? AND i.language = ?',
-            ['spam', BL::getWorkingLanguage()]
+        $repository = BackendModel::get('doctrine.orm.default_entity_manager')
+            ->getRepository(Comment::class);
+
+        $comments = $repository->findBy(
+            [
+                'status' => 'spam',
+                'locale' => BL::getWorkingLanguage(),
```

Comment:
Is the locale actually needed here? It makes sense (to me) to just delete all spam comments, regardless of locale

Reply:
that might result in side-effects, I would leave it like this